pkg/ingester/client/pool.go

```diff
@@ -42,156 +31,12 @@ func (cfg *PoolConfig) RegisterFlags(f *flag.FlagSet) {
 	f.BoolVar(&cfg.HealthCheckIngesters, "distributor.health-check-ingesters", true, "Run a health check on each ingester client during periodic cleanup.")
 }
 
-// Pool holds a cache of grpc_health_v1 clients.
-type Pool struct {
-	services.Service
-
-	cfg     PoolConfig
-	ring    ring.ReadRing
-	factory Factory
-	logger  log.Logger
-
-	sync.RWMutex
-	clients map[string]grpc_health_v1.HealthClient
-}
-
-// NewPool creates a new Pool.
-func NewPool(cfg PoolConfig, ring ring.ReadRing, factory Factory, logger log.Logger) *Pool {
-	p := &Pool{
-		cfg:     cfg,
-		ring:    ring,
-		factory: factory,
-		logger:  logger,
-
-		clients: map[string]grpc_health_v1.HealthClient{},
+func NewPool(cfg PoolConfig, ring ring.ReadRing, factory ring_client.PoolFactory, logger log.Logger) *ring_client.Pool {
```

Comment:
Would it make sense to pass `clients` gauge as a parameter? It's name is `cortex_distributor_ingester_clients` after all, so it should be in distributor. 

Actually, looking at it, I think this entire file should go to distributor package now, since it's completely used from there. What do you think?

Reply:
Agree. Moved to `pkg/distributor`.